eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1323,4 +1323,70 @@ public function getSearchableFieldMapData()
 
         return $statement->fetchAll(\PDO::FETCH_ASSOC);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function updateContentTypeCopyIdentifier($typeId, $baseIdentifier)
```

Comment:
@bdunogier What's your thought? While hitting the mark on user friendliness, it gets rather ugly in code.
Much simpler solution would be that this function is  `function updateContentTypeCopyIdentifier($typeId, $newIdentifier)` and `$newIdentifier = "copy_of_<identifier>_$typeId"`, it will make sure identifier is unique, and shorter than the current remote id stuff, but you will get a int which look a bit strange before you realize its the type id.